Fork PRs fail: actions/checkout refuses fork code under pull_request_target

## Summary

Every PR from a fork currently fails `validate` before any validation runs. `actions/checkout@v4` refuses to check out the PR head, so `index.json` is never on disk and each validation step is skipped.

## Error

From [the run on #345](https://github.com/PeonPing/registry/actions/runs/32432916199):

```
##[error]Refusing to check out fork pull request code from a 'pull_request_target' workflow.
This workflow runs with the base repository's GITHUB_TOKEN, secrets, default-branch cache
scope, and runner access. Fetching and executing a fork's code in that trusted context
commonly leads to "pwn request" vulnerabilities.
```

The job then fails at *Detect new/changed packs* with `FileNotFoundError: [Errno 2] No such file or directory: 'index.json'`, and *Validate index.json schema*, *Validate trust tier*, *Validate pack sources*, *Validate category coverage*, and *Validate audio quality* are all skipped.

## Cause

`.github/workflows/validate-index.yml` triggers on `pull_request_target` and checks out the fork's head:

```yaml
on:
  pull_request_target:
    branches: [main]

    - uses: actions/checkout@v4
      with:
        ref: ${{ github.event.pull_request.head.sha }}
```

`actions/checkout` recently began refusing that combination by default. Nothing in the repo changed — the platform did.

## Scope

Every open fork PR is affected, going back to at least #340 (2026-07-31): #340, #341, #343, #344, #345. #341 is from the org itself, so this isn't specific to outside contributors.

## Suggested fix

Setting `allow-unsafe-pr-checkout: true` would restore the old behaviour, but I'd suggest not doing that here — the workflow holds `contents: write` and `pull-requests: write`, and it runs `.github/scripts/quality-check.py` from the checkout, so a fork could modify that script and have it executed with those permissions. That's the exact scenario the new default is guarding against.

The job doesn't actually need the fork's code — only its `index.json`, as data. Checking out the base and fetching just that file keeps the scripts trusted and fixes the failure:

```yaml
    - uses: actions/checkout@v4
      with:
        fetch-depth: 0

    - name: Fetch the PR's index.json
      env:
        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      run: |
        git fetch origin "pull/${{ github.event.pull_request.number }}/head:pr"
        git show pr:index.json > index.json
```

`git show origin/main:index.json`, which *Detect new/changed packs* already relies on, keeps working since the base is checked out.

## Not affected

Worth noting the fail-closed path works correctly: `Auto-approve` and `Enable auto-merge` both read `verdict.txt` and default to `fail` when it's absent, so nothing gets approved or merged while validation is broken. The failure is a blockage, not a hole.

---

Happy to open a PR with the change if that's useful.

--- a/.github/workflows/validate-index.yml
+++ b/.github/workflows/validate-index.yml
@@ -14,10 +14,18 @@ jobs:
   validate:
     runs-on: ubuntu-latest
     steps:
+      # Check out the BASE repo, not the PR head: this workflow runs under
+      # pull_request_target with contents/pull-requests write, so the scripts it
+      # executes must stay trusted. actions/checkout now refuses a fork head here.
       - uses: actions/checkout@v4
         with:
-          ref: ${{ github.event.pull_request.head.sha }}
           fetch-depth: 0
+
+      # The PR's index.json is data, not code — fetch just that one file.
+      - name: Fetch index.json from the PR
+        run: |
+          git fetch --no-tags origin "pull/${{ github.event.pull_request.number }}/head:pr-head"
+          git show pr-head:index.json > index.json
 
       - name: Validate index.json schema
         id: schema

--- a/.github/workflows/validate-index.yml
+++ b/.github/workflows/validate-index.yml
@@ -25,7 +25,7 @@ jobs:
       - name: Fetch index.json from the PR
         run: |
           git fetch --no-tags origin "pull/${{ github.event.pull_request.number }}/head:pr-head"
-          git show pr-head:index.json > index.json
+          git show pr-head:index.json > /tmp/pr-index.json
 
       - name: Validate index.json schema
         id: schema

--- a/.github/workflows/validate-index.yml
+++ b/.github/workflows/validate-index.yml
@@ -21,11 +21,16 @@ jobs:
         with:
           fetch-depth: 0
 
-      # The PR's index.json is data, not code — fetch just that one file.
+      # The PR's index.json is data, not code, so fetch just that one file and
+      # write it over the base copy so every validation step below reads the
+      # PR's index. Writing it anywhere else leaves the validators looking at
+      # main, where they would find no new packs and pass without checking the
+      # submission. `git show origin/main:index.json`, which "Detect
+      # new/changed packs" diffs against, still resolves through the ref.
       - name: Fetch index.json from the PR
         run: |
           git fetch --no-tags origin "pull/${{ github.event.pull_request.number }}/head:pr-head"
-          git show pr-head:index.json > /tmp/pr-index.json
+          git show pr-head:index.json > index.json
 
       - name: Validate index.json schema
         id: schema

